wgengine/router/ifconfig_windows.go

```diff
@@ -622,28 +604,51 @@ func unicastIPNets(ifc *winipcfg.IPAdapterAddresses) []*net.IPNet {
 // doing the minimum number of AddAddresses & DeleteAddress calls.
 // This avoids the full FlushAddresses.
 //
-// Any IPv6 link-local addresses are not deleted.
+// Any IPv6 link-local addresses are not deleted out of caution as some
+// configurations may repeatedly re-add them. Link-local addresses are adjusted
+// to apply SkipAsSource to prevent them from being used, a the addresses will
+// not be routed on the Tailscale interface.
 func syncAddresses(ifc *winipcfg.IPAdapterAddresses, want []*net.IPNet) error {
-	var erracc error
+	var erracc []error
 
 	got := unicastIPNets(ifc)
 	add, del := deltaNets(got, want)
-	del = excludeIPv6LinkLocal(del)
+
+	ll := make([]*net.IPNet, 0)
 	for _, a := range del {
+		// do not delete link-local addresses, and collect them for later
+		// applying SkipAsSource.
+		if isIPv6LinkLocal(a) {
+			ll = append(ll, a)
+			continue
+		}
+
 		err := ifc.LUID.DeleteIPAddress(*a)
 		if err != nil {
-			erracc = fmt.Errorf("deleting IP %q: %w", *a, err)
+			erracc = append(erracc, fmt.Errorf("deleting IP %q: %w", *a, err))
 		}
 	}
 
 	for _, a := range add {
 		err := ifc.LUID.AddIPAddress(*a)
 		if err != nil {
-			erracc = fmt.Errorf("adding IP %q: %w", *a, err)
+			erracc = append(erracc, fmt.Errorf("adding IP %q: %w", *a, err))
```

Comment:
I'm a little concerned that these were bounded before and now we could have a ton of them in the multierror.

How about returning at most one per type (del, add, get-ll, set-ll)?

Reply:
I don't think we expect a lot of addresses on these interfaces, or a lot of errors.

If errors do come up, it seemed more useful for diagnosis to have more information, the code before was simply replacing the error, dropping all but the last.

The error accumulation is unrelated to the intent of the change, I was just trying to cleanup on the way past, but I'm not really clear on what is useful here - I don't know if one per type is more or less useful for diagnosis than just having the last error.

I think I'll continue to go with diff minimization so we can get the user issue addressed promptly. I've reverted the accumulation.

Reply:
Oh, I was thinking of O(n peers) but right, this is our self addresses. Nevermind.